go/vt/vtgate/planbuilder/operators/system_tables.go

```diff
@@ -37,60 +37,66 @@ func (r *Route) findSysInfoRoutingPredicatesGen4(predicates []sqlparser.Expr, re
 			continue
 		}
 
+		if r.SysTableTableName == nil {
+			r.SysTableTableName = map[string]evalengine.Expr{}
+		}
+
 		if isTableSchema {
 			r.SysTableTableSchema = append(r.SysTableTableSchema, out)
 		} else {
-			if r.SysTableTableName == nil {
-				r.SysTableTableName = map[string]evalengine.Expr{}
-			}
 			r.SysTableTableName[bvName] = out
 		}
 	}
 	return nil
 }
 
 func extractInfoSchemaRoutingPredicate(in sqlparser.Expr, reservedVars *sqlparser.ReservedVars) (bool, string, evalengine.Expr, error) {
```

Comment:
Perhaps adding a comment here to explain what the return parameters are would be 👌🏻